### ISSUE TITLE
Fieldtype Grid

### ISSUE THREAD

**Issue by bognarl** [aryehdesigns/statamic-fetch#4]
Hi. I just realized that if I use Grid type field in my collection Fetch only returns the last item of the array because of the `$array->collapse()` in `Fetch.php:47`.

**This is what it looks like in the md:**
```
output_fields:
  - 
    name: Some name
    cell: G2
  - 
    name: Some other name
    cell: H2
  - 
    name: Some other other name
    cell: I2
```

**And this is what Fetch returns:**
```
  "output_fields": {
    "name": "Some other other name",
    "cell": "I2"
  },
```

Do you have any idea on how to prevent this? I don't want to remove `$array->collapse()` as it is really useful on other fields.

**Comment by aryehdesigns** [comment]
Hmm interesting, I hadn't tested Fetch using a Grid fieldtype. Would need to run some tests to find a solution to this data structure, nothing comes to mind right now.

**Comment by bognarl** [comment]
I was wondering if it is possible to examine fieldtype, but it looks like there is no API for that. Or I did not find it. I am thinking about a solution like this:

```
    public function goDeep($item)
    {
        return collect($item)->map(function ($value, $key) {
            if (is_array($value)) {
                $array = collect($value)->map(function ($value) {
                    return $this->goDeep($value);
                });
                if ($fieldtype == 'Grid') {
                    return $array;
                } else {
                    return $this->isAssoc($array->toArray()) ? $array : $array->collapse();
               }
            }

            return $this->isValid($value, $key) ? $this->relatedData($value) : $value;
        });
    }
```

**Comment by aryehdesigns** [comment]
Right, that would work if the fieldtype was returned, but entries are read directly from the yaml so that won't work unfortunately.

**Comment by aryehdesigns** [comment]
Hi @bognarl, I just released v2.0 of Fetch which fixes this bug and adds support for fetching Page data. Feel free to re-open if your issue persists.

**Comment by bognarl** [comment]
Hi @aryehdesigns, thanks for the update. I'll take a look on that.
